### PR TITLE
Add initial draft of a Roadmap page

### PR DESCRIPTION
**Pull request by djhoese**
A couple months ago a couple of us have been having "design meetings" to work towards a Pyresample 2.0 design. As part of these meetings we conducted a user survey and got a lot of good responses. After a lot of back and forth we ended up with some important changes and what we think are good interfaces. In order to get the word out there and get more feedback on this I've put together a Roadmap document for the website that people (maintainers and users) can refer to when talking about Pyresample 2.0.

Let me know what you think. I'm not sure how much of this I just assumed from our meetings and how much is what we actually agreed upon. You won't hurt my feelings if you completely disagree. That's the point of this, let's discuss.

### CODE REVIEW

File: docs/source/roadmap.rst [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+From our survey we learned that a lot of users use the indexes returned by
+``get_neighbour_info`` for their own custom analyses. We recognize this need
+and while Cache objects could be written to get the same result, we think
+there is a better way. We'd like to explore the idea of "Index" generating
+objects. Like Resamplers, these would provide you a way of relating a source
+geometry to a destination geometry. However, these objects would only be
+responsible for returning the index information.
+
+We haven't defined the interface for these yet, but hope that having something
+separate from resamplers will serve more people.
```
Comment (mraspaud) [inline docs/source/roadmap.rst:162]: I'd like to add here something to make clear that the feature will be there. Right now I understand that we will do exploration but it doesn't sound like we are sure that the feature will be there.

So maybe:
```
From our survey we learned that a lot of users use the indexes returned by
``get_neighbour_info`` for their own custom analyses. We recognize this need
and will implement it. While Cache objects could be written to get the same result, [...]
```

**Review by mraspaud — COMMENTED** [review]
Thanks for putting this together! It looks really good. I just have a small comment about the last paragraph.

File: docs/source/roadmap.rst (hunk 1 above)
Comment (djhoese) [reply docs/source/roadmap.rst:162]: I see what you're saying. Let me see if I can reword it a little bit more.

Comment (djhoese) [reply docs/source/roadmap.rst]: Ok, latest commit should address this and make it a little clearer...I hope.

Comment (mraspaud) [reply docs/source/roadmap.rst]: works

**Comment by mraspaud** [comment]
The rtd build is failing, is that expected?

**Comment by djhoese** [comment]
No, but I think this was fixed in @adybbroe's PR. I may have made this branch without updating to include his changes (I had this roadmap sitting locally for a while).

**Comment by djhoese** [comment]
Or I'll do it now :wink:

**Comment by djhoese** [comment]
Coveralls upload failed it seems so I'm restarting the jobs.

File: docs/source/roadmap.rst [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+You'll also be allowed to provide arbitrary metadata to swath definitions and
+the other geometry types.
+
+Resamplers
+^^^^^^^^^^
+
+Currently there are different interfaces for calling the different resampling
+options in Pyresample. Some times you call a function, sometimes you create a
+class and call a "resample" method on it, and sometimes if you want finer
```
Comment (sfinkens) [inline docs/source/roadmap.rst:82]: if you want finer _control_ I guess?

**Review by sfinkens — APPROVED** [review]
Great job, thank you very much! Just one little typo.

File: docs/source/roadmap.rst (hunk 2 above)
Comment (djhoese) [reply docs/source/roadmap.rst]: Thanks.

**Comment by djhoese** [comment]
I'm realizing I should add something about xarray and geoxarray. I'll try to do that later today.

**Comment by djhoese** [comment]
You two approved this before going on vacation. I need to merge this so that I can link to it from a progress report due next week. I can fix any issues you have with it later. Thanks for understanding.

**Comment by sfinkens** [comment]
Now it's even better :)

**Pull request merged**